src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IVariableDeclaration.cs

```diff
@@ -665,6 +672,650 @@ void M1()
+                                        SyntaxFactory.SingleVariableDesignation(SyntaxFactory.Identifier("z"))),
+                                    SyntaxFactory.LiteralExpression(
+                                        SyntaxKind.NumericLiteralExpression, SyntaxFactory.Literal("42", 42)))
+                                ))))));
+            syntaxTree = syntaxTree.GetCompilationUnitRoot().ReplaceNode(rankSpecifierOld, rankSpecifierNew).SyntaxTree;
+
+            string expectedOperationTree = @"
+IVariableDeclarationOperation (1 declarators) (OperationKind.VariableDeclaration, Type: null, IsInvalid) (Syntax: 'int[yswitch{intz=>42}] x')
+  Ignored Dimensions(1):
+      ISwitchExpressionOperation (1 arms) (OperationKind.SwitchExpression, Type: System.Int32, IsInvalid) (Syntax: 'yswitch{intz=>42}')
+        Value: 
+          ILocalReferenceOperation: y (OperationKind.LocalReference, Type: System.Int32, IsInvalid) (Syntax: 'y')
+        Arms(1):
+            ISwitchExpressionArmOperation (1 locals) (OperationKind.SwitchExpressionArm, Type: null, IsInvalid) (Syntax: 'intz=>42')
+              Pattern: 
+                IDeclarationPatternOperation (OperationKind.DeclarationPattern, Type: null, IsInvalid) (Syntax: 'intz') (InputType: System.Int32, DeclaredSymbol: System.Int32 z, MatchesNull: False)
+              Value: 
+                ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 42, IsInvalid) (Syntax: '42')
+              Locals: Local_1: System.Int32 z
+  Declarators:
+      IVariableDeclaratorOperation (Symbol: System.Int32[] x) (OperationKind.VariableDeclarator, Type: null) (Syntax: 'x')
+        Initializer: 
+          null
+  Initializer: 
+    null
+";
+
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // file.cs(6,13): warning CS0219: The variable 'y' is assigned but its value is never used
+                //         int y = 10;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "y").WithArguments("y").WithLocation(6, 13),
+                // file.cs(7,22): error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)
+                //         /*<bind>*/int[yswitch{intz=>42}] x;/*</bind>*/
+                Diagnostic(ErrorCode.ERR_ArraySizeInDeclaration, "[yswitch{intz=>42}]").WithLocation(7, 22),
+                // file.cs(7,42): warning CS0168: The variable 'x' is declared but never used
+                //         /*<bind>*/int[yswitch{intz=>42}] x;/*</bind>*/
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "x").WithArguments("x").WithLocation(7, 42)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<VariableDeclarationSyntax>(new []{syntaxTree}, expectedOperationTree, expectedDiagnostics);
```

Comment:
> VerifyOperationTreeAndDiagnosticsForTest<VariableDeclarationSyntax>(new []{syntaxTree}, expectedOperationTree, expectedDiagnostics); [](start = 11, length = 133)

Does this test fail if we do not visit ```size``` in LocalBinderFactory?

Reply:
Yes